server.go

```diff
@@ -163,21 +163,41 @@ func (s *server) UpdatePin(ctx context.Context, request v0.UpdatePinRequestObjec
 		user = usersResponse.Payload[0]
 	}
 
+	if len(s.cfg.Nuki.SmartLockDevices) == 0 && s.cfg.Nuki.SmartLockDevice == 0 {
+		return nil, fmt.Errorf("no smart lock device configured")
+	}
+	// Retro compatibility
```

Comment:
I think we should remove this out of the handler and always use the new array.

When we load the config, we can append the deprecated field to the new array field and issue a warning about the usage of a deprecated option.